libpromises/expand.c

```diff
@@ -632,7 +632,7 @@ void BundleResolvePromiseType(EvalContext *ctx, const Bundle *bundle, const char
             for (size_t i = 0; i < SeqLength(pt->promises); i++)
             {
                 Promise *pp = SeqAt(pt->promises, i);
-#if 1
+#if 0
```

Comment:
interesting. Variable re-definition *always* occurs to my knowledge since years now. policy => "free/overrideable" has no effect. (though pre-eval is quite cloudy). The only way to stop variable re-definiton in the current language is to guard the definition of the variable with isvariable(). The other case is I think for cached functions. Probably useless comment, but I wanted to share what i knew. So it's interesting that there is some guard against variable re-definition during pre-eval.